TrackletAlgorithm/SWLUTReader.h

```diff
@@ -0,0 +1,31 @@
+#include <fstream>
+#include <string>
+// The following function provides a more compiler efficient method for reading
+//look up table arrays (currently used in TP), this is only used in C simulations and not synthesis
+template<class lutType, int lutsize> bool readSWLUT(lutType lut[lutsize], std::string lutpath, bool skipfirst = true, bool hex = false){
+
+  //std::cout << "I'm reading : "<< lutpath <<std::endl;
+
+  std::string emdatapath =
+#ifndef __CSIM__ //at run time the working directory differs between CSIM and future SW
```

Comment:
Using "ifndef" here is unnecessarily obscure. Use "ifdef" instead and reverse the order of the two lines inside the if else endif statement.